src/components/Folder.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { getBurstErrors } from '../lib/input-helpers';
+import { saveCurrentGraph, loadSavedGraph } from '../lib/calc-helpers';
+import { getSavedGraphsList } from '../lib/local-storage-helpers';
+import panes from '../constants/pane-types';
+import './Folder.css';
+
+class Folder extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: '',
+      errors: {}
+    };
+
+    this.handleInputUpdate = this.handleInputUpdate.bind(this);
+    this.handleSaveCurrent = this.handleSaveCurrent.bind(this);
+    this.handleLoadGraph = this.handleLoadGraph.bind(this);
+  }
+
+  handleInputUpdate(evt) {
+    const {
```

Comment:
A good place to use the callback pattern for setState if you're trying to access the current state. Also, clean up destructuring. Something like: 

```
const { name, value } = evt.target;
this.setState(st => {
  const { errors, ...nextState } = st;
  nextState[name] = value;
  nextState.errors = getBurstErrors(nextState);
  return nextState;
});
```

Reply:
Done